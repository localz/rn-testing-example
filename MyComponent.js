import React from 'react'
import { View, Text } from 'react-native'

const MyText = 'Hello World! This is my component :D'

export default class MyComponent extends React.Component {
  render() {
    return (
      <View>
        <Text>
          {MyText}
        </Text>
      </View>
    )
  }
}