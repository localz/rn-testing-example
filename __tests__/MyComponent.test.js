import React from 'react'
import renderer from 'react-test-renderer'
import MyComponent from '../MyComponent'

const init = props => {
  const snapshot = renderer.create(<MyComponent {...props} />)
  return {
    snapshot: snapshot.toJSON(),
    renderer: snapshot.getInstance()
  }
}

describe('My Component', () => {
  it('should render properly', () => {
    const { snapshot } = init()
    expect(snapshot).toMatchSnapshot()
  })
})
