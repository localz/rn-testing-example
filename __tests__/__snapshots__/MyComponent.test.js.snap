// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component should render properly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello World! This is my component :D
  </Text>
</View>
`;
